rust/main/agents/relayer/src/msg/pending_message.rs

```diff
@@ -711,3 +711,40 @@ impl MessageSubmissionMetrics {
             .set(std::cmp::max(self.last_known_nonce.get(), msg.nonce as i64));
     }
 }
+
+#[cfg(test)]
+mod test {
+    use std::time::Duration;
+
+    use crate::msg::pending_message::PendingMessage;
+
+    #[test]
+    fn test_calculate_msg_backoff_non_decreasing() {
+        let mut cumulative = Duration::from_secs(0);
+        let mut last_backoff = Duration::from_secs(0);
+
+        for i in 0..100 {
+            let backoff_duration =
+                PendingMessage::calculate_msg_backoff(i).unwrap_or(Duration::from_secs(0));
+            // Useful for showing the impact of changes to the backoff duration
+            println!(
```

Comment:
`println` usually slow down the testing pipeline but `100` seems small enough. Or do you intend to keep this test around as `#[ignore]`d?

Reply:
Went down to 50, but agree a bit ugly - just commenting it out for now so it can be used if needed